src/java.base/share/classes/sun/net/www/protocol/file/FileURLConnection.java

```diff
@@ -46,6 +47,12 @@ public class FileURLConnection extends URLConnection {
     private static final String TEXT_PLAIN = "text/plain";
     private static final String LAST_MODIFIED = "last-modified";
 
+    // The feature of falling back to FTP for non-local file URLs is disabled
+    // by default and can be re-enabled by setting a system property
+    private static boolean FTP_FALLBACK_ENABLED = "true".equalsIgnoreCase(
+            System.getProperty("sun.net.www.protocol.file.ftp-enabled", "false")
+    );
+
```

Comment:
The code for defining and checking this system property would ideally belong in `Handler.java`.

I decided to put it in `FileURLConnection.java` instead, just to avoid increasing the already significant duplication of code between unix/windows implementations of `Handler.java`.